### DISCUSSION TITLE
Incorrect class_type requirements in the specification

### DISCUSSION THREAD

**Discussion by YohDeadfall** [dotnet/csharplang#2753] [Q&A] [answered]
From [the specification](https://github.com/dotnet/csharplang/blob/master/spec/classes.md#type-parameter-constraints):

> A class_type constraint must satisfy the following rules:
>
> * The type must not be one of the following types: System.Array, System.Delegate, System.Enum, or System.ValueType.

This is incorrect since C# 7.3 which allows to specify `Enum` and `Delegate` as a class type constraint.

**Comment by svick** [comment, marked as answer]
As [the spec index](https://github.com/dotnet/csharplang/blob/master/spec/README.md) says, this is the spec for C# 6.0. As I understand it, update of the spec to C# 7.x is planned, but it's probably going to take some time.

In the meantime, I don't think it makes sense to update the spec to include bits and pieces of C# 7.x, since that would make it very hard to understand what the spec is actually describing.

**Comment by YohDeadfall** [comment]
Oh, thanks, missed that. It's strange that the spec is too outdated (C# 8.0 is coming soon).
